tests/integration/test_trigger.py

```diff
@@ -71,3 +50,199 @@ async def test_trigger_no_flows(one, start, log_filter):
+        assert one.pool._get_flow_nums([4, 5]) == set([4, 5])
+        # the only active task still only has flows {1, 2}
+        assert one.pool._get_flow_nums([FLOW_ALL]) == set([1, two])
+
+
+async def test_trigger(flow, scheduler, start):
+    """Test flow assignment when triggering tasks.
+
+    Active tasks:
+      By default keep existing flows, else merge with requested flows.
+    Inactive tasks:
+      By default assign active flows; else assign requested flows.
+
+    """
+    conf = {
+        'scheduler': {
+            'allow implicit tasks': 'True'
+        },
+        'scheduling': {
+            'graph': {
+                'R1': "foo & bar => a & b & c & d & e"
+            }
+        }
+    }
+    id_ = flow(conf)
+    schd = scheduler(id_, run_mode='simulation', paused_start=True)
+    async with start(schd):
+        active_a, active_b = schd.pool.get_tasks()
+        schd.pool.merge_flows(
+            active_b, schd.pool._get_flow_nums([FLOW_NEW]))
+        assert active_a.flow_nums == set([1])
+        assert active_b.flow_nums == set([1, 2])
+
+        #-----(1. Test active tasks)-----
+
+        # By default active tasks keep existing flow assignment.
+        schd.pool.force_trigger_tasks(
+            [active_a.identity], flow=[]
+        )
+        assert active_a.flow_nums == set([1])
+
+        # Else merge existing flow with requested flows.
+        schd.pool.force_trigger_tasks(
+            [active_a.identity], flow=[FLOW_ALL]
+        )
+        assert active_a.flow_nums == set([1, 2])
+
+        # (no-flow is ignored for active tasks)
+        schd.pool.force_trigger_tasks(
+            [active_a.identity], flow=[FLOW_NONE]
+        )
+        assert active_a.flow_nums == set([1, 2])
+
+        schd.pool.force_trigger_tasks(
+            [active_a.identity], flow=[FLOW_NEW]
+        )
+        assert active_a.flow_nums == set([1, 2, 3])
+
+        #-----(2. Test inactive tasks)-----
+
+        # By default inactive tasks get all active flows.
+        schd.pool.force_trigger_tasks(
+            ['1/a'], flow=[]
+        )
+        assert schd.pool._get_task_by_id('1/a').flow_nums == set(
+            [1, 2, 3]
+        )
+
+        # Else assign requested flows.
+        schd.pool.force_trigger_tasks(
+            ['1/b'], flow=[FLOW_NONE]
+        )
+        assert schd.pool._get_task_by_id('1/b').flow_nums == set([])
+
+        schd.pool.force_trigger_tasks(
+            ['1/c'], flow=[FLOW_NEW]
+        )
+        assert schd.pool._get_task_by_id('1/c').flow_nums == set([4])
+
+        schd.pool.force_trigger_tasks(
+            ['1/d'], flow=[FLOW_ALL]
+        )
+        assert schd.pool._get_task_by_id('1/d').flow_nums == set(
+            [1, 2, 3, 4]
+        )
+        schd.pool.force_trigger_tasks(
+            ['1/e'], flow=[7]
+        )
+        assert schd.pool._get_task_by_id('1/e').flow_nums == set([7])
+
+
+async def test_set(flow, scheduler, start):
```

Comment:
I've combined this with the `trigger` test to ensure the behaviours are tested identically for both: https://github.com/hjoliver/cylc-flow/pull/57